Update Git & GitHub course materials (#118)
vibbits/training-material: topics/git-introduction/tutorials/1_introduction/tutorial.md

- Why should I version control and how to get started? 
- How does Git work in its essence? 
objectives:

> - Understand why version controlling is essential in research projects

Reviewer: ```suggestion
- Understand why version control is essential in research projects
```

# 1. Introduction

> Have you also been in a similar and recognizable situation as depicted below? Saving different versions of your files and scripts is essential to keep track of changes, though it can become chaotic very quicly if we do not manage it properly. This is what Git does for us essentially. It works similar to [Google Docs'](https://support.google.com/drive/answer/2409045?co=GENIE.Platform%3DDesktop&hl=en) history feature in which Google automatically saves your document and the changes that happened at a particular moment in time. However, Git allows you to control and decide yourself when changes are worth saving, hence making it much more powerful and flexible. Each change is saved together with a message that enables you or your collaborators to keep an overview of the history of the project.  

Reviewer: ```suggestion
Have you also been in a similar and recognizable situation as depicted below? Saving different versions of your files and scripts is essential to keep track of changes, though it can become chaotic very quickly if we do not use the excellent tools we have available to us. Git is one of these excellent tools. It works similar to [Google Docs'](https://support.google.com/drive/answer/2409045?co=GENIE.Platform%3DDesktop&hl=en) history feature in which Google automatically saves your document and the changes that happened at a particular moment in time. However, Git allows you to control and decide yourself when changes are worth saving, hence making it much more powerful and flexible. Each change is saved together with a message that enables you or your collaborators to keep an overview of the history of the project.  
```

---

> Git is an open-source tool that manages your project (files) by keeping track of each version of these files throughout its history. 

Reviewer: ```suggestion
Git is an open-source tool that keeps track of the changes made to your project files throughout their history. 
```

Why should you version control? 
- **Keeping track of changes** to your files done by yourself or your collaborators. At any moment you can exploit the history of the project to see who wrote what on a particular day. It even allows you to go back to a specific version or undo specific edits. 
- **Synchronizes files between different people or infrastructures** (i.e. laptops, servers, ...), making it a powerful collaborating system. 

> - **Testing new code/changes**. Git can control multiple side versions of the same project in which you can make some changes and only when you or your collaborators are happy with hem, you can include them in the main version.

Reviewer: ```suggestion
- **Testing new code/changes**. Git can control multiple alternative versions of the same project in which you can make some changes and only when you or your collaborators are happy with hem, you can include them in the main version.
```
